packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -103,9 +104,26 @@ export type CollectedDependencies<+TSplitCondition> = $ReadOnly<{
 
 // Registry for the dependency of a module.
 // Defines when dependencies should be collapsed.
-// E.g. should a module that's once required optinally and once not
-// be tretaed as the smae or different dependencies.
+// E.g. should a module that's once required optionally and once not
+// be treated as the same or different dependencies.
 export interface ModuleDependencyRegistry<+TSplitCondition> {
+  /**
+   * Given an import qualifier, return a key used to register the dependency.
+   * Generally this return the `ImportQualifier.name` property, but in the case
+   * of `require.context` more attributes can be appended to distinguish various combinations that would otherwise conflict.
+   *
+   * For example, the following case would have collision issues if they all utilized the `name` property:
+   * ```
+   * require('./foo');
+   * require.context('./foo');
+   * require.context('./foo', true, /something/);
+   * require.context('./foo', false, /something/);
+   * require.context('./foo', false, /something/, 'lazy');
+   * ```
+   *
+   * This method should be utilized by `registerDependency`.
+   */
+  getKeyForDependency(qualifier: ImportQualifier): string;
```

Comment:
I would not expose `getKeyForDependency` in the registry interface; it's already part of the `registerDependency` method's responsibilities so it should just happen internally. If anything does need to know the key `collectDependencies` derived for a dependency, we will likely bake it into the `InternalDependency` data structure rather than expose a separate method.

Reply:
(I get that this is exposed at least in part for testing, but see my comment on the dependency registry tests)

Reply:
I would still prefer it if you turned this into a local function in the file instead of a method on the interface, and tested its behaviour via `collectDependencies`. Having `getKeyForDependency` on the interface will require us to update all the implementations of `ModuleDependencyRegistry`, including the mock (which is failing CI) and the one we have internally at Meta (which will delay landing this). This makes the abstraction unnecessarily leaky (and harder to delete, which as I've mentioned I want to do soon).